### PR TITLE
Align spring-cloud-serveless to main

### PR DESCRIPTION
**Pull request by kelwinpa**
- update to spring boot 3.0.1
- update to java 17
- remove `repo.spring.io`

### CODE REVIEW

File: spring-cloud-serverless/README.md [hunk 1]
```diff
@@ -41,44 +45,3 @@ You can access the app using `curl`:
 ```bash
 curl -w'\n' -H 'Content-Type: text/plain' localhost:8080 -d "Fun"
 ```
-
-## Deploying to Azure Spring Apps with Azure CLI
```
Comment (maly7) [inline spring-cloud-serverless/README.md:45]: Why remove the ASA instructions from the README?

**Comment by maly7** [comment]
Since this is now targeting Java 17, the `az spring app deploy` command in the readme should include the build-env option for Java 17: `--build-env BP_JVM_VERSION=17`

File: spring-cloud-serverless/kubernetes/tap/DEPLOYING.md [hunk 2]
```diff
@@ -44,6 +44,17 @@ tanzu apps workload create hello-fun -f config/workload.yaml \
...
 
+### Deploying to Kubernetes as a TAP workload using GraalVM native compile
+
+If you would like to build the serverless app using the GraalVM native compile feature, then all you need to do is set the build env `BP_NATIVE_IMAGE` to true in the `workload.yaml` before applying it.
+
+This can also be accomplished using the Tanzu CLI with the following command:
+
+```
+tanzu apps workload create hello-fun -f config/workload.yaml \
+  --build-env "BP_NATIVE_IMAGE=true"
```
Comment (maly7) [inline spring-cloud-serverless/kubernetes/tap/DEPLOYING.md:55]: ASA also supports building native images, we should include similar instructions in the ASA deployment instructions.

File: spring-cloud-serverless/accelerator.yaml [hunk 3]
```diff
@@ -3,88 +3,29 @@ accelerator:
...
-        text: Java 11
-      - value: "17"
-        text: Java 17
-    defaultValue: "11"
-    required: true
+    - java
+    - spring
+    - cloud
+    - serverless
+    - tanzu
```
Comment (kvmw) [inline spring-cloud-serverless/accelerator.yaml:10]: the "tanzu" tag  can be removed, i believe.

File: spring-cloud-serverless/accelerator.yaml [hunk 4]
```diff
@@ -3,88 +3,29 @@ accelerator:
...
       onConflict: Append
       sources:
-      - include: [ "README.md"]
-        chain:
-        - type: ReplaceText
-          substitutions:
-          - text: hello-fun template repo
-            with: "#artifactId"
-
-  # Java Version
```
Comment (kvmw) [inline spring-cloud-serverless/accelerator.yaml:52]: Why the build section has been removed? (Maven/Gradle)

**Pull request merged**